src/components/transactions/TxFilterButton/index.tsx

```diff
@@ -1,31 +1,48 @@
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
 import Button from '@mui/material/Button'
 import ExpandLessIcon from '@mui/icons-material/ExpandLess'
 import ExpandMoreIcon from '@mui/icons-material/ExpandMore'
 import FilterAltOutlinedIcon from '@mui/icons-material/FilterAltOutlined'
 import Box from '@mui/material/Box'
+import { useRouter } from 'next/router'
+import { getFilterlessQuery, hasTxFilterQuery } from '../TxFilterForm/utils'
+import TxFilterForm, { TxFilterFormFieldNames } from '@/components/transactions/TxFilterForm'
 
-import TxFilterForm from '@/components/transactions/TxFilterForm'
+const TxFilterButton = ({ className }: { className?: string }) => {
+  const router = useRouter()
 
-import css from './styles.module.css'
-
-const TxFilterButton = () => {
   const [showFilter, setShowFilter] = useState(false)
 
   const toggleFilter = () => {
     setShowFilter((prev) => !prev)
   }
 
+  const hasFilter = hasTxFilterQuery(router.query)
+
+  // Remove filter query when leaving the tx history list
+  useEffect(() => {
+    return () => {
+      if (!hasFilter) {
+        return
+      }
+      router.replace({
+        pathname: router.pathname,
+        query: getFilterlessQuery(router.query),
+      })
+    }
+  }, [])
```

Comment:
This shouldn't be necessary. Nav links should never sent you to a new path with the same query.